Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -78,58 +78,81 @@ public class GameAPIRepo: GameRepository {
     
   }
   
-  //TODO: Replace gameList with individual game and make the model call the function many times.
   public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
     
-    let dispatchGroup = DispatchGroup()
+    let operationQueue = OperationQueue()
     
-    gameList.forEach({ (game) in
-      
-      dispatchGroup.enter()
+    let serviceCallOperation = createServiceCallBlockOperation(using: serviceCaller,
+                                                               forGamesList: gameList)
+    
+    let notifyOperation = BlockOperation {
+      self.notifyGameDetailsLoaded(withData: gameList)
+    }
+    
+    notifyOperation.addDependency(serviceCallOperation)
+    
+    operationQueue.addOperation(serviceCallOperation)
+    operationQueue.addOperation(notifyOperation)
+    
+  }
+  
+  private func createServiceCallBlockOperation(using serviceCaller: ServiceCaller,
+                                               forGamesList gameList: [Game]) -> BlockOperation {
+    
+    let serviceCallOperation = BlockOperation {
       
-      let dataBundle = DataBundle()
-      dataBundle.extraData["game"] = game
+      let dispatchGroup = DispatchGroup()
       
-      // MARK: - Call Succeeded Closure
-      serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+      gameList.forEach({ (game) in
+        
+        dispatchGroup.enter()
         
-        guard let game = dataBundle.extraData["game"] as? Game else {
-          return
+        let dataBundle = DataBundle()
+        dataBundle.extraData["game"] = game
```

Comment:
I would store this "game" string in a struct somewhere as we use it in a few places. Would avoid issues if you misspelt or 'mis-capitalized' it and you can change it in one place if needed.